Library/Homebrew/software_spec.rb

```diff
@@ -208,6 +208,8 @@ def recursive_requirements
   end
 
   def patch(strip = :p1, src = nil, &block)
+    gpatch_dep = gpatch_dep_if_needed(:build)
+    dependency_collector.add(dep) if gpatch_dep
```

Comment:
```suggestion
    dependency_collector.add(gpatch_dep) if gpatch_dep
```

Reply:
@sjackman OK, so I did weapons grade dumb things to my original fork.  So I blew it away and reforked the project.  Then I made the changes you recommended:

`brew/Library/Homebrew/dependency_collector.rb
84-  def java_dep_if_needed(tags)
85-    JavaRequirement.new(tags)
86-  end
87-
88:  def self.gpatch_dep_if_needed(tags)
89:    Dependency.new("gpatch", tags) unless which("patch")
90-  end

brew/Library/Homebrew/software_spec.rb
207-    Requirement.expand(self)
208-  end
209-
210-  def patch(strip = :p1, src = nil, &block)
211:    gpatch_dep = gpatch_dep_if_needed(:build)
212:    dependency_collector.add(gpatch_dep) if gpatch_dep
213-    p = Patch.create(strip, src, &block)
214-    dependency_collector.add(p.resource) if p.is_a? ExternalPatch
215-    patches << p
216-  end
`

Now I am getting entirely new problem behaviour:

`/opt/TundraWare>brew install screen
Error: undefined method `include?' for :build:Symbol
Please report this bug:
  https://docs.brew.sh/Troubleshooting
/opt/TundraWare/tools/Library/Homebrew/dependable.rb:23:in `test?'
/opt/TundraWare/tools/Library/Homebrew/dependency.rb:91:in `block in expand'
/opt/TundraWare/tools/Library/Homebrew/vendor/portable-ruby/2.3.7/lib/ruby/2.3.0/delegate.rb:341:in `each'
/opt/TundraWare/tools/Library/Homebrew/vendor/portable-ruby/2.3.7/lib/ruby/2.3.0/delegate.rb:341:in `block in delegating_block'
/opt/TundraWare/tools/Library/Homebrew/dependency.rb:87:in `expand'
/opt/TundraWare/tools/Library/Homebrew/formula_installer.rb:439:in `expand_requirements'
/opt/TundraWare/tools/Library/Homebrew/formula_installer.rb:384:in `compute_dependencies'
/opt/TundraWare/tools/Library/Homebrew/formula_installer.rb:148:in `verify_deps_exist'
/opt/TundraWare/tools/Library/Homebrew/formula_installer.rb:141:in `prelude'
/opt/TundraWare/tools/Library/Homebrew/cmd/install.rb:331:in `install_formula'
/opt/TundraWare/tools/Library/Homebrew/cmd/install.rb:264:in `block in install'
/opt/TundraWare/tools/Library/Homebrew/cmd/install.rb:262:in `each'
/opt/TundraWare/tools/Library/Homebrew/cmd/install.rb:262:in `install'
/opt/TundraWare/tools/Library/Homebrew/brew.rb:102:in `<main>'
`

Reply:
```
Error: undefined method `include?' for :build:Symbol
```
A Ruby Symbol does not have an `include?` method, but a String does, and a Symbol can be converted to a String using the method `to_s`.

```
❯❯❯ irb
irb(main):001:0> symbol = :build
=> :build
irb(main):002:0> symbol.include? "build"
NoMethodError: undefined method `include?' for :build:Symbol
	from (irb):2
	from /usr/bin/irb:11:in `<main>'
irb(main):003:0> symbol.to_s.include? "build"
=> true
```

Reply:
That's a red herring though. The problem in this case is that `gpatch_dep_if_needed` is passing a symbol, and we're expecting an Array of symbols.